engine/schema/src/com/cloud/storage/dao/StoragePoolTagsDaoImpl.java

```diff
@@ -77,4 +92,71 @@ public void deleteTags(long poolId) {
         txn.commit();
     }
 
+    @Override
+    public List<StoragePoolTagVO> searchByIds(Long... stIds) {
+        final int detailsBatchSize = getDetailsBatchSize();
+
+        // query details by batches
+        List<StoragePoolTagVO> uvList = new ArrayList<StoragePoolTagVO>();
+        int curr_index = 0;
+
+        if (stIds.length > detailsBatchSize) {
```

Comment:
Ah, no this is not what I meant.
I meant only removing the `if` and letting the `while` directly there.
Just deleting line 103